qabelbox/src/main/java/de/qabel/qabelbox/persistence/RepositoryFactory.java

```diff
@@ -108,6 +109,11 @@ public AndroidClientDatabase getAndroidClientDatabase() throws SQLException {
         if (connection == null) {
             connection = DriverManager.getConnection("jdbc:sqlite:" + getDatabasePath());
             androidClientDatabase = new AndroidClientDatabase(connection);
+            try {
+                androidClientDatabase.migrate();
+            } catch (MigrationException e) {
+                throw new RuntimeException(e);
```

Comment:
This case leaves a connection and androidClientDatabase unmigrated. The next call to getAndroidClientDatabase() would return an androidClientDatabase without exception that is probably useless (because it is probably not fully migrated). I would prefer dropping the androidClientDatabase and retry migration on the next `getAndroidClientDatabase()` call. That would either work or throw an exception but not return an unmigrated instance